src/app/CommandSender.h

```diff
@@ -95,35 +88,10 @@ class CommandSender final : public Messaging::ExchangeDelegate
          * @param[in] aAdditionalResponseData
          *                            Additional response data that comes within the InvokeResponseMessage.
          */
-        virtual void OnResponseWithAdditionalData(CommandSender * apCommandSender, const ConcreteCommandPath & aPath,
-                                                  const StatusIB & aStatusIB, TLV::TLVReader * apData,
-                                                  const AdditionalResponseData & aAdditionalResponseData)
-        {
-            OnResponse(apCommandSender, aPath, aStatusIB, apData);
-        }
+        virtual void OnResponse(CommandSender * apCommandSender, const ConcreteCommandPath & aPath,
+                                const StatusIB & aStatusIB, TLV::TLVReader * apData,
+                                const AdditionalResponseData & aAdditionalResponseData) {}
 
-        /**
-         * OnResponse will be called when a successful response from server has been received and processed. Specifically:
-         *  - When a status code is received and it is IM::Success, aData will be nullptr.
-         *  - When a data response is received, aData will point to a valid TLVReader initialized to point at the struct container
-         *    that contains the data payload (callee will still need to open and process the container).
-         *
-         * The CommandSender object MUST continue to exist after this call is completed. The application shall wait until it
-         * receives an OnDone call to destroy the object.
-         *
-         * It is advised that subclasses should only override this or `OnResponseWithAdditionalData`. But, it shouldn't actually
-         * matter if both are overridden, just that `OnResponse` will never be called by CommandSender directly.
-         *
-         * @param[in] apCommandSender The command sender object that initiated the command transaction.
-         * @param[in] aPath           The command path field in invoke command response.
-         * @param[in] aStatusIB       It will always have a success status. If apData is null, it can be any success status,
-         *                            including possibly a cluster-specific one. If apData is not null it aStatusIB will always
-         *                            be a generic SUCCESS status with no-cluster specific information.
-         * @param[in] apData          The command data, will be nullptr if the server returns a StatusIB.
-         */
-        virtual void OnResponse(CommandSender * apCommandSender, const ConcreteCommandPath & aPath, const StatusIB & aStatusIB,
```

Comment:
So I assume this is being removed to get people to notice that the contract changed?

This is fine with my Darwin hat on, but seems pretty hostile for C++ consumers.  Have you run this by @tcarmelveilleux ?

Reply:
i think it would be fine with the android side.